### PR TITLE
Upgrade kcd-scripts

### PR DESCRIPTION
**Pull request by alexandernanberg**
**What**:
Upgrades `kcd-scripts` and makes necessary changes (mostly linting)

**Why**:
Keeping up-to-date

**How**:
Upgrade dep and run `npm run validate` to find  eslint errors etc. 

**Checklist**:

- [ ] Documentation N/A
- [ ] Tests N/A
- [x] Ready to be merged 
      
- [ ] Added myself to contributors table N/A

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -92,6 +92,7 @@
   "eslintConfig": {
     "extends": "./node_modules/kcd-scripts/eslint.js",
     "rules": {
+      "import/no-useless-path-segments": "off",
```
Comment (alexandernanberg) [inline package.json]: Your fix on `eslint-config-kentcdodds` didn't publish (CI failing) so I added an temporary rule until the publish

File: test/basic.test.js [hunk 2]
```diff
@@ -21,11 +21,13 @@ export default class App extends React.Component<Props, State> {
   }
 
   onChange = (selectedItem: Item) => {
```
Comment (alexandernanberg) [inline test/basic.test.js]: Unsure what the original code even did, but I guess an console.log is fine?

File: test/basic.test.js [hunk 3]
```diff
@@ -21,11 +21,13 @@ export default class App extends React.Component<Props, State> {
...
-    selectedItem
+    // eslint-disable-next-line no-console
+    console.log('selectedItem', selectedItem)
   }
 
   onUserAction = (changes: StateChangeOptions<Item>) => {
```
Comment (alexandernanberg) [inline test/basic.test.js]: `onUserAction` isn't used anywhere AFAIK, so it should be safe to remove that?

File: test/custom.test.js [hunk 4]
```diff
@@ -36,7 +36,8 @@ export default class App extends React.Component<Props, State> {
   }
 
   onChange = (selectedItem: Item) => {
```
Comment (alexandernanberg) [inline test/custom.test.js]: Same here, can't see that it's being used somewhere?

File: test/basic.test.js (hunk 3 above)
Comment (kentcdodds) [reply test/basic.test.js]: I think this file is used for TypeScript tests, let's leave it :+1:

File: test/custom.test.js (hunk 4 above)
Comment (kentcdodds) [reply test/custom.test.js]: Used for TypeScript, so I think we can leave it. I kinda leave this stuff up to the TS maintainers.

**Comment by alexandernanberg** [comment]
Yea I really like it, it's a bit bloated with the preact bundles though but I don't think it's worth doing anything about it right now. 

Sure, will do 👍

**Comment by alexandernanberg** [comment]
I've added a note about the size snapshot file. Writing good docs isn't my strong side so feel free to edit it if you like 😅 Wasn't sure where it made most sense to place it either, but I think it kind of works where it is right now.

**Pull request merged**